homeassistant/components/hvac/zwave.py

```diff
@@ -0,0 +1,231 @@
+    value.set_change_verified(False)
+    add_devices([ZWaveHvac(value)])
+    _LOGGER.debug('discovery_info=%s and zwave.NETWORK=%s',
+                  discovery_info, zwave.NETWORK)
+
+
+# pylint: disable=too-many-arguments
+class ZWaveHvac(ZWaveDeviceEntity, HvacDevice):
+    """Represents a HeatControl hvac."""
+
+    # pylint: disable=too-many-public-methods, too-many-instance-attributes
+    def __init__(self, value):
+        """Initialize the zwave hvac."""
+        from openzwave.network import ZWaveNetwork
+        from pydispatch import dispatcher
+        ZWaveDeviceEntity.__init__(self, value, DOMAIN)
+        self._node = value.node
+        self._target_temperature = None
+        self._current_temperature = None
+        self._current_operation = None
+        self._operation_list = None
+        self._current_operation_state = None
+        self._current_fan_mode = None
+        self._fan_list = None
+        self._current_swing_mode = None
+        self._swing_list = None
+        self._unit = None
+        self._zxt_120 = None
+        self.update_properties()
+        # register listener
+        dispatcher.connect(
+            self.value_changed, ZWaveNetwork.SIGNAL_VALUE_CHANGED)
+        # Make sure that we have values for the key before converting to int
+        if (value.node.manufacturer_id.strip() and
+                value.node.product_id.strip()):
+            specific_sensor_key = (int(value.node.manufacturer_id, 16),
+                                   int(value.node.product_id, 16),
+                                   value.index)
+
+            if specific_sensor_key in DEVICE_MAPPINGS:
+                if DEVICE_MAPPINGS[specific_sensor_key] == WORKAROUND_ZXT_120:
+                    _LOGGER.debug('Found Remotec ZXT-120 Zwave Thermostat')
+                    self._zxt_120 = 1
+
+    def value_changed(self, value):
+        """Called when a value has changed on the network."""
+        if self._value.node == value.node:
+            self.update_properties()
+            self.update_ha_state(True)
+            _LOGGER.debug('Value changed on network %s)', value)
+
+    def update_properties(self):
+        """Callback on data change for the registered node/value pair."""
+        # Set point
+        for _, value in self._node.get_values(class_id=0x43).items():
+            if int(value.data) != 0:
+                self._target_temperature = int(value.data)
+        # Operation Mode
+        for _, value in self._node.get_values(class_id=0x40).items():
+            self._current_operation = value.data
+            self._operation_list = list(value.data_items)
+            _LOGGER.debug('self._operation_list=%s)', self._operation_list)
+        # Current Temp
+        for _, value in self._node.get_values(class_id=0x31).items():
+            self._current_temperature = int(value.data)
+            self._unit = value.units
+        # Fan Mode
+        if self._zxt_120 == 1:
```

Comment:
Since the code for 0x42 is equal to 0x44, wouldn't it make sense to just do something as:

```python
fan_class_id = 0x44 if self._zxt_120 else 0x42

for value in self._node.get_values(class_id=fan_class_id).values():
    …
```